### PR TITLE
first push commit

### PR DESCRIPTION
**Pull request by Aquariax**

### CODE REVIEW

**Review by LorenzoEvans — COMMENTED** [review]
## Code Review Checklist

- [x] Follows Lambda Git Flow 
- [x] Frequent, clear, and descriptive commits
- [x] No errors in the console
- [x] Code is written by the student
- [x] Code is sufficiently commented
- [x] Code is organized and easy to read
- [x] MVP Complete


### Notes:
> This is some really awesome code, the createPosts/Tags functions are really impressive, great use of other concepts such as seeding and middleware as well.

File: common/validateUserNameMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      return res.status(400).json({
+        errorMessage: 'Please provide a name for the user.'
+      });
+    } else if (userName.length > 128) {
+      return res.status(400).json({
+        errorMessage: 'Name too long. Use up to 128 characters.'
+      });
+    } else if (userName[0] !== userName[0].toUpperCase()) {
+      return res.status(400).json({
+        errorMessage: 'First letter of name must be uppercase.'
```
Comment (BrannanC) [inline common/validateUserNameMiddleware.js:14]: Very strict 😅

File: data/helpers/postDb.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      query
+        .join('users as u', 'p.userId', 'u.id')
+        .select('p.text', 'u.name as postedBy')
+        .where('p.id', id);
+
+      const promises = [query, this.getPostTags(id)]; // [ posts, tags ]
+
+      return Promise.all(promises).then(function(results) {
+        let [posts, tags] = results;
+        let post = posts[0];
```
Comment (BrannanC) [inline data/helpers/postDb.js:17]: Wow, you really went all out for this project

File: data/migrations/20180404090411_bootstrap.js [hunk 3]
```diff
@@ -1,27 +1,116 @@
...
+
+        console.log('tags table created');
+        resolve(knex);
+      })
+      .catch(error => reject(error));
+  });
+}
+
+function createPostTagsTable(knex) {
+  console.log('creating posttags table');
```
Comment (BrannanC) [inline data/migrations/20180404090411_bootstrap.js:92]: This is really nice database modeling

File: posts/postRouter.js [hunk 4]
```diff
@@ -1,27 +1,80 @@
...
+    .catch(err => res.status(500).json({ error: err }));
 });
 
+// /api/posts/:id
 router.get('/:id', (req, res) => {
+  const id = req.params.id;
+
+  db.get(id)
+    .then(user => {
+      if (!user) {
```
Comment (BrannanC) [inline posts/postRouter.js:23]: this is `post` not user, right?

**Review by BrannanC — COMMENTED** [review]
I think some of the requirements changed since you completed this project and includes more middleware. I'm not too worried about it, though, you clearly know how to use middleware and absolutely crushed this assignment.

**Pull request merged**